uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/ssh/GitSSHService.java

```diff
@@ -51,26 +55,64 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(GitSSHService.class);
 
-    private final SshServer sshd = buildSshServer();
-
+    private static List<BuiltinCiphers> managedCiphers =
+            Collections.unmodifiableList(Arrays.asList(
+                    BuiltinCiphers.aes128ctr,
+                    BuiltinCiphers.aes192ctr,
+                    BuiltinCiphers.aes256ctr,
+                    BuiltinCiphers.arcfour256,
+                    BuiltinCiphers.arcfour128,
+                    BuiltinCiphers.aes192cbc,
+                    BuiltinCiphers.aes256cbc
+            ));
+
+    private static List<BuiltinMacs> managedMACs =
+            Collections.unmodifiableList(Arrays.asList(
+                    BuiltinMacs.hmacmd5,
+                    BuiltinMacs.hmacsha1,
+                    BuiltinMacs.hmacsha256,
+                    BuiltinMacs.hmacsha512,
+                    BuiltinMacs.hmacsha196,
+                    BuiltinMacs.hmacmd596
+            ));
+
+    private SshServer sshd;
     private FileSystemAuthenticator fileSystemAuthenticator;
     private FileSystemAuthorizer fileSystemAuthorizer;
     private SSHAuthenticator sshAuthenticator;
 
-    private static SshServer buildSshServer() {
-        final List<BuiltinCiphers> ciphers =
-                Collections.unmodifiableList(Arrays.asList(
-                        BuiltinCiphers.aes128ctr,
-                        BuiltinCiphers.aes192ctr,
-                        BuiltinCiphers.aes256ctr,
-                        BuiltinCiphers.arcfour256,
-                        BuiltinCiphers.arcfour128,
-                        BuiltinCiphers.aes192cbc,
-                        BuiltinCiphers.aes256cbc
-                ));
+    private SshServer buildSshServer() {
+        return builder().cipherFactories(
+                setUpBuiltinFactories(false,
+                                      managedCiphers)).
+                macFactories(
+                        setUpBuiltinFactories(false,
+                                              managedMACs)).build();
+    }
 
-        return builder().cipherFactories(setUpBuiltinFactories(false,
-                                                               ciphers)).build();
+    private SshServer buildSshServer(List<BuiltinCiphers> ciphersConfigured,
+                                     List<BuiltinMacs> macsConfigured) {
+        ServerBuilder serverBuilder = builder();
+        if(!ciphersConfigured.isEmpty()){
```

Comment:
this code should be simplified, the parameter should be a predicate to filter the list, so we don't have to mantain 2 lists and check if input is empty.

Reply:
ok